Automattic/tracks-chrome-extension
# Improve link handling

This PR refactors the link display logic to add more links and to fix a bug where anonymous users would have a Store Admin link. The full list of changes in this PR is:
 * Bump version to 1.6
 * Change CSS class for links from `.store-admin` to `.content-link`
 * Move the link rendering into a dedicated `renderLink()` function
 * Refactor `generateProperties()` to use `map()` and to work out whether we have a logged-in user id
 * Refactor `generateProperty()` to achieve multiple goals:
    - Make it clearer which fields get special handling, both via comments and by switching to an `if/else if/else` structure
    - When we have a logged-in user, add links to Store Admin and the User Report Card (and remove the leading whitespace inside the link)
    - Add a link to Blog RC for blog_id

#### Screenshots

##### Anonymous user

<img width="1265" alt="Screenshot 2022-08-15 at 20 22 15" src="https://user-images.githubusercontent.com/3376401/184694136-b6fb73aa-d40d-4173-aac3-209d7c1bb0de.png">

##### Logged-in user

<img width="1265" alt="Screenshot 2022-08-15 at 20 26 56" src="https://user-images.githubusercontent.com/3376401/184694148-3fd51657-3555-4ee6-98c9-6adda40ec590.png">

#### Testing Instructions

* Ensure that you have a copy of the `Tracks Vigilante` folder from the `improve/link-handling` branch on your local machine -- you can do that by downloading it from GitHub, or by cloning the repo and checking out the `improve/link-handling` branch
* Enable development mode in the Chome Extensions configuration
* Click on the "Load unpacked" button, and select the location of your `Tracks Vigilante` folder
* Once the extension has been uploaded, you need to perform two sets of tests:
  1. While logged in, navigate to `wordpress.com/home` and select a site
    - Open Tracks Vigilante and either open the standalone popup, or check the "Extended properties" checkbox
    - Verify that `blog_id` has a valid Blog RC link
    - Verify that `_ul` and `_ui` have valid links to Store Admin and User RC that load the correct user
    - Verify that `_dl` and `_dr` load the correct pages when you click on them too
  2. While logged out (which you can achieve by logging out, or enabling Tracks Vigilante in incognito mode and then opening an incognito window), navigate to `wordpress.com/start`
    - Open Tracks Vigilante and either open the standalone popup, or check the "Extended properties" checkbox
    - Verify that the `_ui` item does _not_ have any links next to it
    - Verify that clicking on the `_dl` and `_dr` items (when present) load the correct pages
